themes/default/content/registry/packages/azure-native/from-v2-to-v3/_index.md

```diff
@@ -0,0 +1,250 @@
+
+Review your program for warnings on missing imports or deprecated resources. If your program uses any deprecated explicit API versions where the import contains the API version, you will need to update these since explicit API versions are no longer included in the SDK. It's recommnded to migrate to the default version, if possible, or you can generate a local SDK for the desired API version as described in the [version guide](./version-guide).
+
+### Upgrade Dependencies
+
+In your Pulumi program, upgrade the package to point to the latest v3.x version.
+
+* JavaScript/TypeScript: [`@pulumi/azure-native`](https://www.npmjs.com/package/@pulumi/azure-native/v/3.0.0)
+* Python: [`pulumi-azure-native`](https://pypi.org/project/pulumi-azure-native/3.0.0/)
+* Go: [`github.com/pulumi/pulumi-azure-native/sdk/go/azure`](https://github.com/pulumi/pulumi-azure-native/releases/tag/v3.0.0)
+* .NET: [`Pulumi.AzureNative`](https://www.nuget.org/packages/Pulumi.AzureNative/3.0.0)
+* Java: [`com.pulumi.azurenative`](https://central.sonatype.com/artifact/com.pulumi/azure-native/3.0.0)
+
+{{< chooser language "typescript,python,csharp,go" >}}
+
+{{% choosable language typescript %}}
+
+```
+-"@pulumi/azure-native": "^2.0.0",
+- "resolved": "https://registry.npmjs.org/@pulumi/azure-native/-/azure-native-2.89.3.tgz",
+- "resolved": "https://registry.npmjs.org/@pulumi/azure-native/-/azure-native-2.89.3.tgz",
++ @pulumi/azure-native": "v3.0.0",
++ "resolved": "https://registry.npmjs.org/@pulumi/azure-native/-/azure-native-3.0.0.tgz",
++ "resolved": "https://registry.npmjs.org/@pulumi/azure-native/-/azure-native-3.0.0.tgz",
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+- pulumi_azure_native==v2.89.3
++ pulumi_azure_native==v3.0.0
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+- <PackageReference Include="Pulumi.AzureNative" Version="2.89.3"
++ <PackageReference Include="Pulumi.AzureNative" Version="3.0.0"
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+- github.com/pulumi/pulumi-azure-native-sdk/storage v2.89.3
++ github.com/pulumi/pulumi-azure-native-sdk/v3/storage v3.0.0
+```
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+### Upgrade Imports
+
+Go programs will need all imports updated to include `v3` in the path.
+
+```go
+import (
+- "github.com/pulumi/pulumi-azure-native-sdk/resources/v2"
+- "github.com/pulumi/pulumi-azure-native-sdk/storage/v2"
++ "github.com/pulumi/pulumi-azure-native-sdk/resources/v3"
++ "github.com/pulumi/pulumi-azure-native-sdk/storage/v3"
+)
+```
+
+If you do not replace the deprecated import, you can expect to see an error message:
+
+```bash
+    error: Running program '<path/to/project>' failed with an unhandled exception:
+Error: Cannot find module '@pulumi/azure-native/resources/v20210501'
+```
```

Comment:
should we stick this inside the language chooser?

Reply:
Actually, I'm not 100% sure that works, but if it does it would be nice since other languages don't need this info